compiler/src/dotty/tools/backend/jvm/DottyBackendInterface.scala

```diff
@@ -853,8 +853,12 @@ class DottyBackendInterface(outputDirectory: AbstractFile, val superCallsMap: Ma
 
     def addRemoteRemoteExceptionAnnotation: Unit = ()
 
-    def samMethod(): Symbol =
-      toDenot(sym).info.abstractTermMembers.headOption.getOrElse(toDenot(sym).info.member(nme.apply)).symbol
+    def samMethod(): Symbol = {
+      val abstractTermMembers = toDenot(sym).info.abstractTermMembers
+      if (abstractTermMembers.size > 1) abort(s"${sym.show} is not a functional interface. " +
```

Comment:
I would pattern-match on `abstractTermMembers` and have cases for it being empty, one element and multiple elements.